torch_frame/data/dataset.py

```diff
@@ -335,15 +342,17 @@ def materialize(self, device: Optional[torch.device] = None,
 
         # 1. Fill column statistics:
         for col, stype in self.col_to_stype.items():
-            if stype == torch_frame.multi_categorical:
-                ser = self.df[col].apply(
-                    lambda x: [cat.strip() for cat in x.split(self.sep)]
-                    if x is not None and x != '' else [])
-                ser = ser.explode()
-                ser = ser.dropna()
-            else:
-                ser = self.df[col]
-            self._col_stats[col] = compute_col_stats(ser, stype)
+            ser = self.df[col]
+            if stype == torch_frame.multicategorical and (self.sep is None or
+                                                          col not in self.sep):
```

Comment:
```suggestion
            if (stype == torch_frame.multicategorical and (self.sep is None or
                                                          col not in self.sep)):
```
will give you a better formatting. +1 no reason to check for `self.sep is None` if you do `self.sep = sep or {}` in constructor (sorry if this wasn't clear).